addons/waterways/river_gizmo.gd

```diff
@@ -213,17 +422,42 @@ func commit_handle(gizmo: EditorSpatialGizmo, index: int, restore, cancel: bool
 	ur.add_undo_property(river, "valid_flowmap", river.valid_flowmap)
 	ur.add_undo_method(river, "update_configuration_warning")
 	ur.commit_action()
-	
+
+	_extra_lines = {}
+
 	redraw(gizmo)
 
+func draw_vertical_line_to_terrain(index, pos, global_pos, space_state, material):
+	var worldpos = pos + global_pos
+	var result = space_state.intersect_ray(worldpos, worldpos + Vector3.DOWN * 128)
+	if not result:
+		result = space_state.intersect_ray(worldpos, worldpos + Vector3.UP * 128)
+	if result:
+		var ground_pos_relative = result.position - global_pos + Vector3(0, abs(result.position.y - global_pos.y)/2, 0)
+		var line_length = (pos - ground_pos_relative).length()
+		_extra_lines[index] = {}
+		_extra_lines[index]["mat"] = material
+		_extra_lines[index]["p1"] = pos
+		_extra_lines[index]["p2"] = ground_pos_relative
+		_extra_lines[index+100000] = {}
+		_extra_lines[index+100000]["mat"] = material
+		_extra_lines[index+100000]["p1"] = ground_pos_relative - Vector3(1, 0, 0) * line_length * 0.1
+		_extra_lines[index+100000]["p2"] = ground_pos_relative + Vector3(1, 0, 0) * line_length * 0.1
```

Comment:
Why are some of these index offset by 100000? Again I'm sure I could reason it out, but I'd like a comment explaining it since it's not immediate clear what's happening here.